core/src/components/footer/footer.tsx

```diff
@@ -46,6 +51,30 @@ export class Footer implements ComponentInterface {
     this.checkCollapsibleFooter();
   }
 
+  connectedCallback() {
+    if (typeof (window as any) !== 'undefined') {
```

Comment:
Can we avoid casting to `any` here? I know there are a few places in core where we have a similar check out casting; although those may be `.ts` files instead of `.tsx`. 

Edit: This looks like maybe the result of copy & paste from the tab bar component. I do not think the type cast is necessary here (or in tab bar).

Reply:
This is an SSR check to make sure that we do not access `window` when it is not available. However, it would be better to use https://github.com/ionic-team/ionic-framework/blob/main/core/src/utils/window/index.ts so that we do not need to keep doing this check everywhere.